src/scalar/text.ts

```diff
@@ -0,0 +1,58 @@
+import { Utf8 as ArrowString } from '@apache-arrow/esnext-esm';
+
+import { Scalar } from './scalar.js';
+import { isInvalid, NULL_VALUE } from './util.js';
+
+export class Text implements Scalar<string> {
+  private _valid = false;
+  private _value = '';
+
+  public constructor(v: unknown) {
+    this.value = v;
+    return this;
+  }
+
+  public get dataType() {
+    return new ArrowString();
+  }
+
+  public get valid(): boolean {
+    return this._valid;
+  }
+
+  public get value(): string {
+    return this._value;
+  }
+
+  public set value(value: unknown) {
+    if (isInvalid(value)) {
+      this._valid = false;
+      return;
+    }
+
+    switch (typeof value) {
+      case 'object': {
+        if (value !== undefined && value !== null && Object.hasOwn(value, 'toString')) {
```

Comment:
```suggestion
        if (Object.hasOwn(value, 'toString')) {
```
We already check for invalid `value` above with `isInvalid()`

Reply:
Without this check the typescript compiler/builder gave errors. If there is a way around that I would be happy to change it